Modules/IO/NRRD/src/itkNrrdImageIO.cxx

```diff
@@ -1102,6 +1104,8 @@ void NrrdImageIO::Write(const void *buffer)
        && this->m_NrrdCompressionEncoding->available() )
     {
     nio->encoding = this->m_NrrdCompressionEncoding;
+    nio->zlibLevel = this->GetCompressionLevel();
```

Comment:
It might be this simple. Have you tried writing some images with different compression levels to see whether this gets propagated all the way?

Reply:
I'll check...